chia/wallet/wallet_state_manager.py

```diff
@@ -1196,6 +1202,25 @@ async def handle_clawback(
             await self.tx_store.add_transaction_record(tx_record)
         return None
 
+    async def handle_vc(self, parent_coin_spend: CoinSpend) -> Optional[WalletIdentifier]:
+        # Check the ownership
+        vc: VerifiedCredential = VerifiedCredential.get_next_from_coin_spend(parent_coin_spend)
+        derivation_record: Optional[DerivationRecord] = await self.puzzle_store.get_derivation_record_for_puzzle_hash(
+            vc.inner_puzzle_hash
+        )
+        if derivation_record is None:
+            self.log.warning(
+                f"Verified credential {vc.launcher_id.hex()} is not belong to the current wallet."
+            )  # pragma: no cover
+            return None  # pragma: no cover
+        self.log.info(f"Found verified credential {vc.launcher_id.hex()}.")
+        for wallet_info in await self.get_all_wallet_info_entries(wallet_type=WalletType.VC):
```

Comment:
this looks like it's quickly going to be a performance issue. presumably, *all* of these won't fit in RAM

Reply:
I'm not sure what you are referring. Do you mean we have too many wallets to fit the memory?

Reply:
I think this is not even relevant here. Looks like a change from #15059.